santa/src/Santa.Api/Controllers/MessageController.cs

```diff
@@ -66,82 +67,89 @@ public async Task<ActionResult<Logic.Objects.Message>> GetMessage(Guid chatMessa
         [Authorize(Policy = "create:messages")]
         public async Task<ActionResult<Logic.Objects.Message>> PostMessage([FromBody] ApiMessageModel message)
         {
-            Logic.Objects.Client logicClient = await repository.GetClientByIDAsync(message.messageSenderClientID.GetValueOrDefault() != Guid.Empty ? message.messageSenderClientID.GetValueOrDefault() : message.messageRecieverClientID.GetValueOrDefault());
-            Logic.Objects.Client checkerClient = await repository.GetClientByEmailAsync(User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Email).Value);
-            BaseClient baseCheckerClient = new BaseClient()
-            {
-                clientID = checkerClient.clientID,
-                clientName = checkerClient.clientName,
-                nickname = checkerClient.nickname,
-                email = checkerClient.email,
-                hasAccount = checkerClient.hasAccount,
-                isAdmin = checkerClient.isAdmin
-            };
-
-            // If the logic client and checker client have the same Id and the relationxref is either null or part of of the checked clients assingments list (Or the checker is just an admin overall)
-            if ((logicClient.clientID == checkerClient.clientID && message.clientRelationXrefID != null ? checkerClient.assignments.Any(a => a.clientRelationXrefID == message.clientRelationXrefID) : true) || checkerClient.isAdmin)
+            BaseClient logicBaseClient = await repository.GetBasicClientInformationByID(message.messageSenderClientID.GetValueOrDefault() != Guid.Empty ? message.messageSenderClientID.GetValueOrDefault() : message.messageRecieverClientID.GetValueOrDefault());
+            BaseClient checkerClient = await repository.GetBasicClientInformationByEmail(User.Claims.FirstOrDefault(c => c.Type == ClaimTypes.Email).Value);
+            List<RelationshipMeta> checkerAssignmentInfo = new List<RelationshipMeta>();

```

Comment:
Changed objects to base client objects, and a list of only assignments to better loading times when posting messages